### PR TITLE
34, 87, 88, 91: Spotify Auth and User Persistance

### PR DESCRIPTION
**Pull request by NicSchuler**
Persistance still in mem

### CODE REVIEW

File: build.gradle
Comment (EliasWJMuller) [inline build.gradle:1]: delete dep mongoDB

Comment (EliasWJMuller) [inline build.gradle:1]: remove MongoDB dep

File: README.md [hunk 1]
```diff
@@ -1,5 +1,12 @@
 # SoPra RESTful Service Template FS24
 
+## Group 28: Necessary Additional Setup
+To run the server locally, following things need to be done:
+
+* System Env Variables: Save the clientSecret and the redirectURL in the system environment variables so that it can be extracted by SpotifyService
+* Make sure that your spotify account is listed in the Spotify "App" (maintained by Nicolas)
+* Install mongoDB locally (https://www.mongodb.com/try/download/community)
```
Comment (hs-kim1990) [inline README.md:8]: No need to install mongoDB, as JpaRepository is used

File: src/main/java/ch/uzh/ifi/hase/soprafs24/entity/SpotifyJWT.java
Comment (EliasWJMuller) [inline src/main/java/ch/uzh/ifi/hase/soprafs24/entity/SpotifyJWT.java:1]: Use Lombok

File: src/main/java/ch/uzh/ifi/hase/soprafs24/entity/player.java
Comment (EliasWJMuller) [inline src/main/java/ch/uzh/ifi/hase/soprafs24/entity/player.java:1]: 1. Capitalize Class
2. use Lombok

File: src/main/java/ch/uzh/ifi/hase/soprafs24/rest/dto/AuthPostCodeDTO.java
Comment (EliasWJMuller) [inline src/main/java/ch/uzh/ifi/hase/soprafs24/rest/dto/AuthPostCodeDTO.java:1]: Use Lombok

File: src/main/java/ch/uzh/ifi/hase/soprafs24/rest/dto/AuthTokensDTO.java
Comment (EliasWJMuller) [inline src/main/java/ch/uzh/ifi/hase/soprafs24/rest/dto/AuthTokensDTO.java:1]: Use Lombok

File: src/main/java/ch/uzh/ifi/hase/soprafs24/controller/AuthController.java
Comment (EliasWJMuller) [inline src/main/java/ch/uzh/ifi/hase/soprafs24/controller/AuthController.java:1]: Put Logic into Service

File: src/main/java/ch/uzh/ifi/hase/soprafs24/service/AuthService.java
Comment (EliasWJMuller) [inline src/main/java/ch/uzh/ifi/hase/soprafs24/service/AuthService.java:1]: is Autowired necessary?
Lombok @RequiredArgsContructor

File: src/main/java/ch/uzh/ifi/hase/soprafs24/service/SpotifyService.java
Comment (EliasWJMuller) [inline src/main/java/ch/uzh/ifi/hase/soprafs24/service/SpotifyService.java:1]: Lombok RequiredArgsContructor?
Might require save and flush when working with persistence

File: src/main/resources/application.properties
Comment (EliasWJMuller) [inline src/main/resources/application.properties:1]: remove mongoDB stuff

**Review by EliasWJMuller — CHANGES_REQUESTED** [review]

File: src/main/resources/application.properties [hunk 2]
```diff
@@ -11,6 +11,10 @@ spring.datasource.username=sa
 spring.datasource.password=
 spring.jpa.database-platform=org.hibernate.dialect.H2Dialect
 
+# mongoDB access (local-Console)
```
Comment (hs-kim1990) [inline src/main/resources/application.properties:14]: not necessary

File: src/main/java/ch/uzh/ifi/hase/soprafs24/service/AuthService.java [hunk 3]
```diff
@@ -1,9 +1,59 @@
...
+
+    public User createOrLogin(String spotifyUserId, String username, SpotifyJWT spotifyJWT) {
+        User userToCreateOrLogin = new User();
+        userToCreateOrLogin.setSpotifyUserId(spotifyUserId);
+        userToCreateOrLogin.setUsername(username);
+
+        if (userService.userExists(userToCreateOrLogin)) {
+            return userService.loginUser(userToCreateOrLogin.getSpotifyUserId(), spotifyJWT);
+        } else {
+            userToCreateOrLogin.setUserId(1L);
```
Comment (hs-kim1990) [inline src/main/java/ch/uzh/ifi/hase/soprafs24/service/AuthService.java:54]: with JpaRepository, userId is a generated value.

File: build.gradle [hunk 4]
```diff
@@ -36,6 +36,9 @@ dependencies {
     annotationProcessor 'org.mapstruct:mapstruct-processor:1.3.1.Final'
     testAnnotationProcessor 'org.mapstruct:mapstruct-processor:1.3.1.Final'
 
+    // implemented to use mongoDB
+    implementation 'org.springframework.boot:spring-boot-starter-data-mongodb'
```
Comment (hs-kim1990) [inline build.gradle:40]: no mongo needed

**Pull request merged**